src/Messages/OWSDisappearingMessagesJob.m

```diff
@@ -121,7 +121,7 @@ - (void)runLoop
 
 + (void)setExpirationForMessage:(TSMessage *)message
 {
-    dispatch_async(self.serialQueue, ^{
+    dispatch_sync(self.serialQueue, ^{
```

Comment:
I don't think we want to do this. This queue can get a bit backed up. If we need to make this testable, we could just have the test wait, or add a completion, etc.